app/models/ems_refresh/save_inventory.rb

```diff
@@ -37,7 +37,8 @@ def save_vms_inventory(ems, hashes, target = nil, disconnect = true)
     log_header = "EMS: [#{ems.name}], id: [#{ems.id}]"
 
     disconnects = if target.kind_of?(ExtManagementSystem) || target.kind_of?(Host)
-                    target.vms_and_templates.reload.to_a
+                    # make sure not to delete vms when disconnect flag set to false
+                    disconnect ? target.vms_and_templates.reload.to_a : []
```

Comment:
I still worry that this doesn't fix the issue where the vm is the target, can you do
```
if disconnect && (target.kind_of ... )
elsif disconnect && target.kind_of?(Vm)
end
```
?

Reply:
Actually we do pass [ems](https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh.rb#L105) so for this specific scenario it should be fine. I will update the PR anyway to be sure for other places where it is used.